samples/excel/85-preview-apis/data-types-entity-attribution.yaml

```diff
@@ -0,0 +1,235 @@
+            }
+          };
+
+          return entity;
+        }
+
+        // Get products and product properties. 
+        function getProduct(productID: number): any {
+          return products.find((p) => p.productID == productID);
+        }
+
+        /** Set up Sample worksheet. */ 
+        async function setup() {
+          await Excel.run(async (context) => {
+            context.workbook.worksheets.getItemOrNullObject("Sample").delete();
+            const sheet = context.workbook.worksheets.add("Sample");
+
+            const productsTable = sheet.tables.add("A1:C1", true /*hasHeaders*/);
+            productsTable.name = "ProductsTable";
+
+            productsTable.getHeaderRowRange().values = [["Product", "ProductID", "ProductName"]];
+
+            productsTable.rows.add(
+              null /*add at the end*/,
+              products.map((p) => [null, p.productID, p.productName])
+            );
+
+            sheet.getUsedRange().format.autofitColumns();
+            sheet.getUsedRange().format.autofitRows();
+
+            sheet.activate();
+
+            await context.sync();
+          });
+        }
+
+        /** Default helper for invoking an action and handling errors. */ 
+        async function tryCatch(callback) {
+          try {
+            await callback();
+          } catch (error) {
+            // Note: In a production add-in, you'd want to notify the user through your add-in's UI.
+            console.error(error);
+          }
+        }
+
+        /** Sample JSON product data. */ 
+        const products = [
+          {
+            productID: 1,
+            productName: "Chai",
+            quantityPerUnit: "10 boxes x 20 bags",
+            unitPrice: 18,
+            providerName: "Microsoft",
+            sourceAddress: "https://github.com/OfficeDev/office-js-snippets/blob/main/.github/images/microsoft-logo.png?raw=true",
+            targetAddress: "http://microsoft.com"
+          },
+          {
+            productID: 2,
+            productName: "Chang",
+            quantityPerUnit: "24 - 12 oz bottles",
+            unitPrice: 19,
+            providerName: "Microsoft",
+            sourceAddress: "https://github.com/OfficeDev/office-js-snippets/blob/main/.github/images/microsoft-logo.png?raw=true",
+            targetAddress: "http://microsoft.com"
+          },
+          {
+            productID: 3,
+            productName: "Aniseed Syrup",
+            quantityPerUnit: "12 - 550 ml bottles",
+            unitPrice: 10,
+            providerName: "MSN",
+            sourceAddress: "https://github.com/OfficeDev/office-js-snippets/blob/main/.github/images/msn-logo.png?raw=true",
+            targetAddress: "http://xbox.com"
+          },
+          {
+            productID: 4,
+            productName: "Chef Anton's Cajun Seasoning",
+            quantityPerUnit: "48 - 6 oz jars",
+            unitPrice: 22,
+            providerName: "Xbox",
```

Comment:
Consider using Bing instead of Xbox, if only because the idea of Xbox dealing with seasoning is a bit jarring (get it? because it's sold in jars? 🤣)

Reply:
Lol, agreed, but will leave as Xbox based on offline conversation about logo use case guidance.